homeassistant/components/shelly/__init__.py

```diff
@@ -122,7 +129,8 @@ async def shutdown(self):
         if self._unsub_stop:
             self._unsub_stop()
             self._unsub_stop = None
-        await self.device.shutdown()
+        if self.hass.data[DOMAIN].get(COAP_CONTEXT):
+            await self.hass.data[DOMAIN].pop(COAP_CONTEXT).shutdown()
```

Comment:
If a user uninstalls 1 Shelly device, we call this code. That would now break every other set up Shelly device. Instead, we should keep track how many devices are set up and only if the last one shuts down should we shut down the context.

Reply:
This is called when Home Assistant stops, not when user remove the device. We check if we can shutdown and remove `coap_context` when user remove device [here](https://github.com/home-assistant/core/pull/41905/files#diff-61a6cfd7c02acb522bf310e992f9d8e7500e95cbcbb34cbed8a739ec07e84d5cR153).

Reply:
Then that's a bug! We need to call this when we unload the integration to unsubscribe the stop listener.

Reply:
Sorry for my stupid mistakes. I think now i fixed this mess.